### DISCUSSION TITLE
How is state persisted?

### DISCUSSION THREAD

**Discussion by wsw70** [jamesoff/simplemonitor#1388] [Q&A]
As I am digging into the code to understand how could steal ideas for Monitors and Loggers I have a hard time understanding how state is preserved (persisted) across Simplemonitor restarts.
Could you please just provide a pointer to some relevant code in Github so that I could dig further? Thanks!

EDIT: in addition to understanding the code, it would help me to get rid of a line in the HTML Logger output, some old Monitor instance is still there

**Comment by jamesoff** [comment]
That's an easy one: it's not :)

There's an issue about it #560 and I did start some work on a branch locally a while back but didn't get very far with it - quite a bit of the times I needed it myself at least were handled by the addition of being able to reload the config on-the-fly.

For your edit, do you mean when you remove a monitor from a remote instance, and the main instance doesn't know it's gone?

**Reply by wsw70** [reply]
> That's an easy one: it's not :)

Indeed :) 

> For your edit, do you mean when you remove a monitor from a remote instance, and the main instance doesn't know it's gone?

It was a mistake on my side - I did not realize that the failed test was from a remote instance. Stopping the remote instance fixed the issue (I am moving towards a remote/agentless test so I will not have many local instances anyway)
